src/transformers/pipelines/visual_question_answering.py

```diff
@@ -87,29 +140,109 @@ def __call__(self, image: Union["Image.Image", str], question: str = None, **kwa
+        num_spans = len(encoding["input_ids"])
+
+        if "boxes" in inputs:
+            boxes = inputs["boxes"]
+            bbox = []
+            for batch_index in range(num_spans):
+                for i, s, w in zip(
+                    encoding.input_ids[batch_index],
+                    encoding.sequence_ids(batch_index),
+                    encoding.word_ids(batch_index),
+                ):
+                    if s == 1:
+                        bbox.append(boxes[w])
+                    elif i == self.tokenizer.sep_token_id:
+                        bbox.append([1000] * 4)
+                    else:
+                        bbox.append([0] * 4)
+
+            if self.framework == "tf":
+                # TODO implement
+                pass
+            elif self.framework == "pt":
+                encoding["bbox"] = torch.tensor([bbox])
+
+        # TODO: Handle multiple spans. We'll basically want to duplicate the image features for each span
+        # (we can then also remove this assert)
+        assert len(image_features) == 0 or list(image_features.items())[0].size(0) == list(encoding.items())[0].size(0)
+        encoding.update(image_features)
+
+        # TODO: I think it's cleaner to place the encoding and other context in the dict in separate keys
+        # instead of flat at the top level. I'm happy to undo this though if it's not in line with
+        # other parts of the code.
+        return {
+            "encoding": encoding,
+            "word_ids": [encoding.word_ids(i) for i in range(len(encoding["input_ids"]))],
+            "sequence_ids": [encoding.sequence_ids(i) for i in range(len(encoding["input_ids"]))],
+            "inputs": inputs,
+        }
+
+    def _forward(self, inputs):
+        model_outputs = self.model(**inputs["encoding"])
+        return {"outputs": model_outputs, "inputs": inputs}
+
+    def postprocess(self, result, top_k=5):
+        model_outputs = result["outputs"]
+
+        # TODO: Is there a better way to do this? I tried using
+        # isinstance(model_outputs, SequenceClassifierOutput) but that thinks model_outputs
+        # is transformers.utils.generic.ModelOutput
+        if "logits" in model_outputs:
+            return postprocess_sequence_output(self.model, model_outputs, self.framework, top_k)
+        elif "start_logits" in model_outputs and "end_logits" in model_outputs:
+            return postprocess_qa_output(
+                self.model,
+                model_outputs,
+                result["inputs"]["word_ids"],
+                result["inputs"]["inputs"]["words"],
+                self.framework,
+                top_k,
+            )
         else:
-            raise ValueError(f"Unsupported framework: {self.framework}")
-
-        scores = scores.tolist()
-        ids = ids.tolist()
-        return [{"score": score, "answer": self.model.config.id2label[_id]} for score, _id in zip(scores, ids)]
+            assert False, "Unknown output format"
```

Comment:
Again please raise real exceptions.

Reply:
Will fix. These were meant to be temporary as I was sketching out the code.

Reply:
No worries.